### DISCUSSION TITLE
HttpClient issue.

### DISCUSSION THREAD

**Discussion by kranthikiran1984** [openai/openai-dotnet#280] [Q&A]
My function calling chatCompletion gives me weird httpresponse. I tried the example given in the package examples. I obtained the symbols and tried to debug. The http request with exact http headers and body is successful from postman but is giving BadRequest from package. BasRequest: missing 'model' parameter.  We are using a gateway that talks to openAI API. I tried to post the request to the gateway directly from postman and it is successful. Is there anything that needs to be done? Thank you for your help.

**Comment by joseharriaga** [comment]
Hello, @kranthikiran1984. Is it possible for you to share your code?

**Reply by kranthikiran1984** [reply]
Hello Jose, If I let Fiddler be my web-proxy and start capturing traffic, I do see proper json response. But with fiddler turned off I observed 2 headers Content-Encoding: br and Transfer-Encoding: Chunked. Does the package handles response content decompression? I would like to re-iterate that we use an AWS gateway endpoint to go to openAI endpoint. My app talks to AWS gateway endpoint. This clearly looks like some kind of decryption issue. Thank you so much for responding and looking forward for your thoughts. Thank you!

**Reply by kranthikiran1984** [reply]
Code-wise I literally used the sample code in the documentation.
